### PR TITLE
TCP: Update history file and version number

### PR DESCRIPTION
**Pull request by AniruddhaKanhere**
Description
-----------
This PR updates the version number and updates the `History.txt` file for the +TCP source files.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by gitwallit — DISMISSED** [review]

File: FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/History.txt [hunk 1]
```diff
@@ -1,3 +1,47 @@
+Changes between V2.2.1 and V2.2.2
+	+ Add CBMC proofs for 4 functions.
```
Comment (RichardBarry) [inline FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/History.txt:2]: Folks may not know what CBMC is - so better to say something like "added another 4 memory safety proofs using the CBMC model checker".

File: FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/History.txt [hunk 2]
```diff
@@ -1,3 +1,47 @@
...
+	+ Address ~1200 MISRA violations.
```
Comment (RichardBarry) [inline FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/History.txt:3]: This looks really alarming, but I expect most of the 1200 are the same small change repeated in lots of places.  Best to be a bit more generic and say something like "stricter MISRA compliance", or "updated to conform to additional MISRA rules [then list them].

File: FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/History.txt [hunk 3]
```diff
@@ -1,3 +1,47 @@
...
+	+ Support caching multiple IP addresses per DNS entry in TCP library.
+	+ Security improvements in the TCP library.
```
Comment (RichardBarry) [inline FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/History.txt:5]: Is this in addition to the CBMC work?  Again I would make it either more generic as "continuous security improvements" or more specific such as "Improved security by giving users the option to reject UDP packets with a checksum of zero even though the spec allows it".

File: FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/History.txt [hunk 4]
```diff
@@ -1,3 +1,47 @@
...
+	+ Add additional length check when checksum offloading is used.
```
Comment (RichardBarry) [inline FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/History.txt:6]: "Improve robustness by checking packet lengths in software even if the check has already been performed in the hardware".

File: FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/History.txt [hunk 5]
```diff
@@ -1,3 +1,47 @@
...
+	+ Add ipconfigIP_PASS_PACKETS_WITH_IP_OPTIONS compile time constant to enable
+	  users to drop packets with IP options (IP Options are not supported).
+	+ Introduce ipconfigUDP_PASS_ZERO_CHECKSUM_PACKETS macro which enables the
```
Comment (RichardBarry) [inline FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/History.txt:9]: Say why, as per my previous comment "improve security by allows the user to drop UDP packets that have a checksum of zero even though the UDP specification says zero checksus should be accepted.".

**Review by RichardBarry — CHANGES_REQUESTED** [review]

**Pull request closed without merging**